### ISSUE TITLE
doc_markdown and Katex

### ISSUE THREAD

**Issue by recmo** [rust-lang/rust-clippy#7273]
Lint name: `doc_markdown`

I tried this code:

```rust
    /// * `dx` The sample spacing such that $x_i = i ⋅ dx$.
```

I expected to see this happen: *no clippy warnings*

Instead, this happened: 

```
warning: you should put `x_i` between ticks in the documentation
  --> src/fill_curve.rs:29:46
   |
29 |     /// * `dx` The sample spacing such that $x_i = i ⋅ dx$.
   |                                              ^^^
   |
```

### Meta

- `cargo clippy -V`: `clippy 0.1.54 (8cf990c9 2021-05-15)`
- `rustc -Vv`:
  ```
  rustc 1.54.0-nightly (8cf990c9b 2021-05-15)
  binary: rustc
  commit-hash: 8cf990c9b5c59f25c806fad9f4466f9d6509bbea
  commit-date: 2021-05-15
  host: x86_64-apple-darwin
  release: 1.54.0-nightly
  LLVM version: 12.0.1
  ```

**Comment by flip1995** [comment]
As `rustdoc` doesn't support LaTeX/Katex, it might be better to actually put this whole `$...$` into backticks, rather than let it render as normal text.

So while the span is not ideal/complete, I'd say that this is not a false positive. Extending the span here, without producing more false positives in other cases is really hard to impossible, so I'll close this issue as not planned.
